### PR TITLE
Use prometheus/client_golang and go modules

### PR DESCRIPTION
**Pull request by sconover**
This is part of an effort to move m3 off of the `github.com/m3db/prometheus_*` dependencies. To do that, we need to get rid of them in tally, too.

tally tests pass:

```
go test `go list ./... | grep -v thirdparty` ; echo $?
ok  	github.com/uber-go/tally	(cached)
?   	github.com/uber-go/tally/example	[no test files]
ok  	github.com/uber-go/tally/instrument	(cached)
ok  	github.com/uber-go/tally/m3	(cached)
ok  	github.com/uber-go/tally/m3/customtransports	(cached)
?   	github.com/uber-go/tally/m3/example	[no test files]
?   	github.com/uber-go/tally/m3/thrift	[no test files]
ok  	github.com/uber-go/tally/m3/thriftudp	(cached)
ok  	github.com/uber-go/tally/multi	(cached)
ok  	github.com/uber-go/tally/prometheus	(cached)
?   	github.com/uber-go/tally/prometheus/example	[no test files]
ok  	github.com/uber-go/tally/statsd	(cached)
?   	github.com/uber-go/tally/statsd/example	[no test files]
0
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module github.com/uber-go/tally
+
+go 1.15
+
+require (
+	github.com/cactus/go-statsd-client v3.1.1+incompatible
+	github.com/m3db/prometheus_client_golang v0.8.1 // indirect
+	github.com/m3db/prometheus_client_model v0.1.0 // indirect
+	github.com/m3db/prometheus_common v0.1.0 // indirect
+	github.com/m3db/prometheus_procfs v0.8.1 // indirect
```
Comment (robskillington) [inline go.mod:10]: Hm, should these libraries still be a dependency?

**Review by robskillington — APPROVED** [review]
LGTM other than the questions regarding continued reference to M3 prometheus library forked dependencies

**Comment by mway** [comment]
Is the goal is to merge this upstream versus maintaining a fork?

**Comment by robskillington** [comment]
@mway we're definitely open to merging upstream - I believe the thought was that uber-go/tally would never accept the change given the build complexities of referencing the vanilla prometheus client library (which in past caused a lot of build failures at Uber given that services/applications depended on open source libraries with conflicting versions of the prometheus client library at multiple differing incompatible 0.x versions of the client library).

**Pull request merged**